packages/happy-dom/src/html-serializer/HTMLSerializer.ts

```diff
@@ -0,0 +1,162 @@
+import Element from '../nodes/element/Element.js';
+import * as PropertySymbol from '../PropertySymbol.js';
+import Node from '../nodes/node/Node.js';
+import DocumentType from '../nodes/document-type/DocumentType.js';
+import HTMLTemplateElement from '../nodes/html-template-element/HTMLTemplateElement.js';
+import NodeTypeEnum from '../nodes/node/NodeTypeEnum.js';
+import ProcessingInstruction from '../nodes/processing-instruction/ProcessingInstruction.js';
+import DocumentFragment from '../nodes/document-fragment/DocumentFragment.js';
+import ShadowRoot from '../nodes/shadow-root/ShadowRoot.js';
+import HTMLElementConfig from '../config/HTMLElementConfig.js';
+import HTMLElementConfigContentModelEnum from '../config/HTMLElementConfigContentModelEnum.js';
+import XMLEncodeUtility from '../utilities/XMLEncodeUtility.js';
+
+/**
+ * Serializes a node into HTML.
+ */
+export default class HTMLSerializer {
+	private options: {
+		serializableShadowRoots: boolean;
+		shadowRoots: ShadowRoot[] | null;
+		allShadowRoots: boolean;
+	} = {
+		serializableShadowRoots: false,
+		shadowRoots: null,
+		allShadowRoots: false
+	};
+
+	/**
+	 * Constructor.
+	 *
+	 * @param [options] Options.
+	 * @param [options.serializableShadowRoots] If shadow roots should be serialized.
+	 * @param [options.shadowRoots] Shadow roots to serialize.
+	 * @param [options.allShadowRoots] If all shadow roots should be serialized.
+	 */
+	constructor(options?: {
+		serializableShadowRoots?: boolean;
+		shadowRoots?: ShadowRoot[] | null;
+		allShadowRoots?: boolean;
+	}) {
+		if (options) {
+			if (options.serializableShadowRoots) {
+				this.options.serializableShadowRoots = options.serializableShadowRoots;
+			}
+
+			if (options.shadowRoots) {
+				this.options.shadowRoots = options.shadowRoots;
+			}
+
+			if (options.allShadowRoots) {
+				this.options.allShadowRoots = options.allShadowRoots;
+			}
+		}
+	}
+
+	/**
+	 * Renders an element as HTML.
+	 *
+	 * @param root Root element.
+	 * @returns Result.
+	 */
+	public serializeToString(root: Node): string {
+		switch (root[PropertySymbol.nodeType]) {
+			case NodeTypeEnum.elementNode:
+				const element = <Element>root;
+				const prefix = element[PropertySymbol.prefix];
+				const localName = element[PropertySymbol.localName];
+				const config = HTMLElementConfig[element[PropertySymbol.localName]];
+				const tagName = prefix ? `${prefix}:${localName}` : localName;
+
+				if (config?.contentModel === HTMLElementConfigContentModelEnum.noDescendants) {
+					return `<${tagName}${this.getAttributes(element)}>`;
```

Comment:
I noticed that for AREA element it doesn't delete the element, it just places the element after it. Test case would be:
```
const parser = new DOMParser();
expect(
  parser.parseFromString("<area><span></span></area>", "text/html").body.innerHTML
).toBe("<area><span></span>")
```
For the col element it does something different, instead of placing it AFTER, it places it before.
Test case:
```
const parser = new DOMParser();
expect(
  parser.parseFromString("<table><col><span></span></col></table>", "text/html").body.innerHTML
).toBe("<span></span><table><colgroup><col></colgroup></table>")
```
It seems you need descendantsAfter / descendantsBefore instead of noDescendants.